core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -566,15 +763,16 @@ public void updateTask(TaskResult taskResult) {
     /**
      * @param workflowId ID of the workflow to evaluate the state for
      * @return true if the workflow has completed (success or failed), false otherwise.
+     * @throws ApplicationException If there was an error - caller should retry in this case.
      */
     public boolean decide(String workflowId) {
 
         //If it is a new workflow the tasks will be still empty even though include tasks is true
         Workflow workflow = executionDAO.getWorkflow(workflowId, true);
-        WorkflowDef def = metadataDAO.get(workflow.getWorkflowType(), workflow.getVersion());
+        // FIXME: The workflow could be null.
```

Comment:
In what scenario could it be null?

Reply:
I have just checked on the `executionDAO` implementations and it seems to be fine. 
@gorzell Any extra context you can give on this one?

Reply:
It was probably just being defensive and may have been more related to calling metadata, as I think that might have returned null for an ID that didn't exist.